### ISSUE TITLE
Parsing Gaussian basis sets with more than 9 exponents/coefficients fails (skips function)

### ISSUE THREAD

**Issue by togloss** [lmmentel/chemtools#1]
I've noticed that in parse_gaussian_basis the regular expression has an issue. 

shellre = re.compile(r'^\s*(?P<shells>[SPDFGHILspdfghil]+)\s*(?P<nf>[1-9]+)\s*(?P<scale>\-?\d+\.\d+)')
I think should be
shellre = re.compile(r'^\s*(?P<shells>[SPDFGHILspdfghil]+)\s*(?P<nf>[0-9]+)\s*(?P<scale>\-?\d+\.\d+)')

That fixes the problem for me. Basis sets with 10 exponents are now accepted. 

I was trying to branch and pull request but I'm new to git and can't seem to figure it out.

Tobias

**Comment by lmmentel** [comment]
Hi Tobias,

Thanks for taking the time to report this! I guess I only tested on small basis sets and didn't spot it. I'll try to patch it for you.

If you don't mind me asking could you tell me what is chemtools most useful for in your case?

**Comment by togloss** [comment]
Hi Lukasz,

I use the tool to avoid or detect human error when using basis sets from here http://qchem.pnpi.spb.ru/recp
Unlike basis set exchange, where you can pick your format, this page gives you random data in CFOUR format mostly but not always. Your tool cannot read CFOUR/ACES II but I can manually copy/paste to Gaussian and then translate back and compare as a check.
I recently started working on gold electrodes for electrochemistry and I have to say, the basis set and ecp landscape looks pretty chaotic to me (the confusion may disappear over time, I hope). Your tool might help bring a little more standardization in so we don't waste time troubleshooting. Why should such data be stored in text files anyway?

Thanks
Tobias

**Comment by lmmentel** [comment]
I managed to push a patched version to pypi, v0.9..2. Once you upgrade your issue should be fixed.

Thanks for the context. It seems that adding a CFOUR basis parser would be a nice feature to have. I wish I had a bit more time for this project to add new features.

I worked a bit with developing new basis sets for heavy elements with and without ECPs and this is mainly where `chemtools` comes from. I agree that text files are not the best for this kind of data especially when there are so many different formats. I don't have a good proposal for the persistence standard for the basis set but at least the internal (python) model was quite useful for programmatic operations on basis sets.
